kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java

```diff
@@ -293,16 +293,21 @@ protected void processFormLayout(FormInstance topLevelForm,
             List<String> scriptDataList) {
         FormLayout layout = form.getLayout();
         
+
         if (form.getModel().getClassName() != null && wrapJson) {
-            jsonTemplate                
-                .append("'")
-                .append(form.getModel().getName())
-                .append("' : ")
-                .append("{")
-                .append("'")
-                .append(form.getModel().getClassName())                
-                .append("' : ")
-                .append("{");
+            Optional<FormField> fieldForm = topLevelForm.getFields().stream().filter(e -> e.getNestedForm().equals(form.getId())).findAny();
```

Comment:
could we use this `org.kie.server.services.jbpm.ui.form.render.model.FormInstance.getNestedForm(String)` instead here?

meaning `topLevelForm..getNestedForm(form.getId())`

Reply:
I am lost with this comment. 
fieldForm.getNestedForm().equals(form.getId()) is not equivalent to
topLevelForm..getNestedForm(form.getId())

what exactly you want to change ?

Reply:
wouldn't getNestedForm return the form or null for given formId? U mean to replace the entire stream, filter, etc

Reply:
The idea is to find the binding in the top level form. Each field has the nested form id that belongs to (AFAIK)

Reply:
there should be a resolver that can look through all children forms so might be able to find any that is defined for the top level one